src/Compilers/CSharp/Portable/Symbols/Source/SimpleProgramNamedTypeSymbol.cs

```diff
@@ -212,14 +212,15 @@ protected override MembersAndInitializers BuildMembersAndInitializers(Diagnostic
                 }
             }
 
-            return new MembersAndInitializers(nonTypeNonIndexerMembers: declaration.Declarations.SelectAsArray<SingleTypeDeclaration, Symbol>(singleDeclaration => new SynthesizedSimpleProgramEntryPointSymbol(this, singleDeclaration, diagnostics)),
-                                              staticInitializers: ImmutableArray<ImmutableArray<FieldOrPropertyInitializer>>.Empty,
-                                              instanceInitializers: ImmutableArray<ImmutableArray<FieldOrPropertyInitializer>>.Empty,
-                                              indexerDeclarations: ImmutableArray<SyntaxReference>.Empty,
-                                              staticInitializersSyntaxLength: 0,
-                                              instanceInitializersSyntaxLength: 0,
-                                              isNullableEnabledForInstanceConstructorsAndFields: false,
-                                              isNullableEnabledForStaticConstructorsAndFields: false);
+            return new MembersAndInitializersBuilder();
+        }
+
+        protected override void AddSynthesizedMembers(MembersAndInitializersBuilder builder, DiagnosticBag diagnostics)
+        {
+            foreach (SingleTypeDeclaration singleDeclaration in declaration.Declarations)
+            {
+                builder.NonTypeNonIndexerMembers.Add(new SynthesizedSimpleProgramEntryPointSymbol(this, singleDeclaration, diagnostics));
```

Comment:
>builder.NonTypeNonIndexerMembers.Add(new SynthesizedSimpleProgramEntryPointSymbol(this, singleDeclaration, diagnostics)); [](start = 16, length = 121)

I don't think separating these is the right thing to do. Please keep creation of these symbols where they used to be created from syntax, i.e. in BuildDeclaredMembersAndInitializers. They are "declared" in source by a compilation unit with global statements.